src/WebJobs.Extensions/Extensions/Core/Bindings/ExecutionContextBindingProvider.cs

```diff
@@ -63,10 +60,17 @@ public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
                     throw new ArgumentNullException("context");
                 }
 
-                return BindInternalAsync(new ExecutionContext
+                return BindInternalAsync(CreateContext(context.FunctionInstanceId));
+            }
+
+            private ExecutionContext CreateContext(Guid invocationId)
+            {
+                return new ExecutionContext
                 {
-                    InvocationId = context.FunctionInstanceId
-                });
+                    InvocationId = invocationId,
+                    FunctionName = _parameter.Member.Name,
```

Comment:
This will just be passed as some context that may be consumed by the function. The function name here will match the method name, which will be the function name in the context of WebJobs. Happy to change this to something else, but based on the conversations I've had with @MikeStall and @mathewc, this seemed correct.